jmclient/jmclient/wallet_utils.py

```diff
@@ -1326,17 +1328,31 @@ def wallet_createwatchonly(wallet_root_path, master_pub_key):
 
     password = cli_get_wallet_passphrase_check()
     if not password:
+        jmprint("The passphrase can not be empty", "error")
         return ""
 
-    entropy = FidelityBondMixin.get_xpub_from_fidelity_bond_master_pub_key(master_pub_key)
-    if not entropy:
-        jmprint("Error with provided master pub key", "error")
-        return ""
+    if is_fidelity_bond_wallet:
+        entropy = FidelityBondMixin.get_xpub_from_fidelity_bond_master_pub_key(master_pub_key)
+        if not entropy:
+            jmprint("Error with provided master public key", "error")
+            return ""
+    else:
+        entropy = master_pub_key
     entropy = entropy.encode()
 
-    wallet = create_wallet(wallet_path, password,
-        max_mixdepth=FidelityBondMixin.FIDELITY_BOND_MIXDEPTH,
-        wallet_cls=FidelityBondWatchonlyWallet, entropy=entropy)
+    if is_fidelity_bond_wallet:
+        create_wallet(wallet_path, password,
+            max_mixdepth=FidelityBondMixin.FIDELITY_BOND_MIXDEPTH,
+            wallet_cls=FidelityBondWatchonlyWallet, entropy=entropy)
+    else:
+        if master_pub_key.startswith('ypub'):
+            wallet_cls = SegwitLegacyWatchonlyWallet
+        else:
+            wallet_cls = SegwitWatchonlyWallet
```

Comment:
I don't think this is correct, `native` setting of `joinmarket.cfg` should be taken into account too. This will create native segwit wallet if xpub is given with `native = false` in config. Also we could force `SegwitWatchonlyWallet` if zpub is given, regardless of `native` setting.

Reply:
> This will create native segwit wallet if xpub is given with native = false in config. Also we could force SegwitWatchonlyWallet if zpub is given, regardless of native setting.

To make sure we're on the same page:
- If `zpub` is given, we create `SegwitWatchonlyWallet` regardless of the `native` config.
- If `ypub` is given, we create `SegwitLegacyWatchonlyWallet` regardless of the `native` config.
- If `xpub` is given, we create `SegwitWatchonlyWallet` if `native == true`, and we create `LegacyWatchonlyWallet` if `native == false`.

Is my understanding correct?

Reply:
I'm wondering, do we want to take `segwit` config into consideration as well?

So, if xpub is given, we create SegwitWatchonlyWallet if native == true && segwit == true, we create SegwitWatchonlyWallet if native == false && segwit == true, and we create LegacyWatchonlyWallet if segwit == false

Reply:
Yes, that's how I think this should work.